pkg/cloudprovider/types.go

```diff
@@ -257,6 +257,31 @@ func (ofs Offerings) Cheapest() Offering {
 	})
 }
 
+// MostExpensive returns the most expensive offering from the return offerings
+func (ofs Offerings) MostExpensive() Offering {
+	return lo.MaxBy(ofs, func(a, b Offering) bool {
+		return a.Price > b.Price
+	})
+}
+
+// CapacityType returns the capacity type requirement for an offering, and panics if a capacity type is not set
+func (of Offering) CapacityType() string {
+	req, ok := of.Requirements[v1beta1.CapacityTypeLabelKey]
+	if !ok {
+		panic("failed to get capacity type for offering, capacity type must be definied on all offerings")
+	}
```

Comment:
also nit on spelling
```suggestion
	if !ok {
		panic("failed to get capacity type for offering, capacity type must be defined on all offerings")
	}
```